experiment/api.js

```diff
@@ -0,0 +1,147 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const {utils: Cu} = Components;
+const {SectionsManager} = Cu.import("resource://activity-stream/lib/SectionsManager.jsm", {});
+const {EventManager} = Cu.import("resource://gre/modules/ExtensionCommon.jsm", {});
+
+// Give ActivityStream actions names that are more consistent with webextension
+// event names. Actions added here must also be added to the schema.
+const ActivityStreamActions = {
+  SYSTEM_TICK: "SystemTick",
+  NEW_TAB_LOAD: "NewTabOpened"
+};
+
+class API extends ExtensionAPI { // eslint-disable-line no-unused-vars
+  get sectionOptions() {
+    const manifestOptions = this.extension.manifest.new_tab_section_options;
+    for (const prop in manifestOptions) {
+      // Necessary to avoid overwriting default values with null if no option is
+      // provided in the manifest
+      if (manifestOptions[prop] === null) { manifestOptions[prop] = undefined; }
```

Comment:
It's probably not a good idea to mutate this object in case it causes unexpected side effects.